tests.c

```diff
@@ -316,6 +322,8 @@ void test_suite_4() {
 }
 
 void test_suite_5(void) {
+    double zero = 0.0; /* msvc is silly (workaround for error C2124) */
```

Comment:
I apparently based this checkedC translation off a newer version of the base parson than my previous translation. This is because some of the issues we found over the summer were changed by the parson maintainer between last summer and January and it seemed more sensible to translate the version of parson that was currently being used. Therefore this PR includes a bunch of changes from the original parson maintainer (kgabis), and others folks in the original C version. This change is one of them, as are any semantically interesting changes in the tests.c